File: QuickSort/QuickSort.php
<?php
class QuickSort{
	
	protected $Array;
	
	public function __construct($Array){
		$this->Array = $Array;
	}
	
	private function swap($i, $j){
		$tmp = $this->Array[$j];
		$this->Array[$j] = $this->Array[$i];
		$this->Array[$i] = $tmp;
	}

	private function Partition($lo, $hi){
		$pivot = $this->Array[$hi];
		$i = $lo-1;
		for($j = $lo; $j <= $hi; $j ++){
			if($this->Array[$j] <= $pivot){
				$i++;
				if($i != $j){
					$this->swap($i, $j);
				}
			}
		}
		return $i;
	}

	public function QuickSort($lo, $hi){
		if($lo < $hi){
			$p = $this->Partition($lo, $hi);
			$this->QuickSort($lo, $p-1);
			$this->QuickSort($p+1, $hi);
		}
	}
	
	public function getSort(){
		return implode(" ",$this->Array);
	}
}

$input = readline();
$Array = array_map('intval', explode(' ', $input));
$obj = new QuickSort($Array);
$obj->QuickSort(0, sizeof($Array)-1);
print_r($obj->getSort());
?>

File: QuickSort/QuickSort.java
import java.util.Scanner;
import java.util.Arrays;

class QuickSort {
	private static int[] inputArray;
	
	private static int Partition(int lo, int hi){
		int pivot = inputArray[hi];
		int i = lo-1;
		int tmp;
		for(int j = lo; j <= hi; j ++){
			if(inputArray[j] <= pivot){
				i ++;
				if(i != j){
					tmp = inputArray[j];
					inputArray[j] = inputArray[i];
					inputArray[i] = tmp;
				}
			}
		}
		return i;
	}
	
	private static void QuickSort(int lo, int hi){
		if(lo < hi){
			int p = Partition(lo, hi);
			QuickSort(lo, p-1);
			QuickSort(p+1, hi);
		}
	}
	
	private static String ArrayToString(int[] array){
		String input = "";
		for(int i = 0; i < array.length; i ++){
			input += array[i];
			if(i != array.length-1){
				input += " ";
			}
		}
		return input;
	}
	
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		String[] input = sc.nextLine().split("\\s+");
		inputArray = new int[input.length];
		for(int i = 0; i < input.length; i ++){
			inputArray[i] = Integer.parseInt(input[i]);
		}
		QuickSort(0, inputArray.length-1);
		System.out.println(ArrayToString(inputArray));
	}
}